* Curriculum Number: IE5054
* Course Name: Data Analytics
* Student ID: r13921031
* Subject: Assignment 3

## 🤔 Question 1. 
Given a simple linear regression model
$ 𝑦_𝑖 = 𝛽_0 + 𝛽_1 * 𝑥_𝑖 + 𝜖_𝑖 $  
$ 𝑖 = 1, … , 𝑛$  
$ 𝜖_𝑖 $ is i.i.d. to $ N(𝜇, 𝜎^2).$  
Prove that:  
### ✅ 1-a. $cov(\hat{𝛽_0},\hat{𝛽_1} ) = − \frac{\overline{x} * 𝜎^2 }{ 𝑆_{𝑥𝑥} }$  

## 📚 参考
 - [Mathematics][1] (https://math.stackexchange.com/questions/788574/find-operatornamecov-hat-beta-0-hat-beta-1)
 - [Mathematics][2] (https://math.stackexchange.com/questions/821318/operatornamevar-hat-beta-1-frac-sigma2-sumx-i-barx2-how-t)

## ① 第一歩 
We just use the result on page 30 of "DA02 Regression Analysis.pdf"
$Var[\hat{\beta}_1]=\frac{\sigma^2}{S_{xx}}$

## ② 第二歩 
$Cov(\hat{\beta}_0, \hat{\beta}_1) = Cov(\bar{y} - \hat{\beta}_1 \bar{x}, \hat{\beta}_1)$

$= E[(\bar{y} - \hat{\beta}_1 \bar{x})\hat{\beta}_1] - E[\bar{y} - \hat{\beta}_1 \bar{x}]E[\hat{\beta}_1]$

$= E[\bar{y}\hat{\beta}_1 - \bar{x}\hat{\beta}_1^2] - (\bar{y} - \bar{x}\beta_1)\beta_1$

$= \bar{y}E[\hat{\beta}_1] - \bar{x}E[\hat{\beta}_1^2] - \bar{y}\beta_1 + \bar{x}E[\hat{\beta}_1]E[\hat{\beta}_1]$

$= \bar{y}\beta_1 - \bar{x}E[\hat{\beta}_1^2] - \bar{y}\beta_1 + \bar{x}\beta_1^2$

$= -\bar{x}(E[\hat{\beta}_1^2] - E[\hat{\beta}_1]E[\hat{\beta}_1])$

$= -\bar{x}Var[\hat{\beta}_1]$
$= -\bar{x}\frac{\sigma^2}{S_{xx}}$

## 👍 DONE.
## ⚠️ The following part is just a stpe-by-step derviation for myself, can just ignore it.

$\hat{\beta}_1 = \frac{\sum_{i=1}^{n} (x_i - \bar{x})(y_i - \bar{y})}{\sum_{i=1}^{n} (x_i - \bar{x})^2} = \frac{S_{xy}}{S_{xx}}$

$\hat{\beta}_0 = \bar{y} - \hat{\beta}_1 \bar{x}$

Where:
- $S_{xx} = \sum_{i=1}^{n} (x_i - \bar{x})^2$
- $S_{xy} = \sum_{i=1}^{n} (x_i - \bar{x})(y_i - \bar{y})$

Step 1: Express $\hat{\beta}_0$ in terms of $y_i$
$\hat{\beta}_0 = \bar{y} - \hat{\beta}_1 \bar{x} = \frac{1}{n}\sum_{i=1}^{n} y_i - \hat{\beta}_1 \bar{x}$

Step 2: Calculate the covariance
$Cov(\hat{\beta}_0, \hat{\beta}_1) = E[(\hat{\beta}_0 - E[\hat{\beta}_0])(\hat{\beta}_1 - E[\hat{\beta}_1])]$

Step 3: Since both estimators are unbiased, we have:
$E[\hat{\beta}_0] = \beta_0$ and $E[\hat{\beta}_1] = \beta_1$

Therefore:
$Cov(\hat{\beta}_0, \hat{\beta}_1) = E[(\hat{\beta}_0 - \beta_0)(\hat{\beta}_1 - \beta_1)]$

Step 4: Substitute the expression for $\hat{\beta}_0$
$\hat{\beta}_0 - \beta_0 = \frac{1}{n}\sum_{i=1}^{n} y_i - \hat{\beta}_1 \bar{x} - \beta_0$

Since $y_i = \beta_0 + \beta_1 x_i + \epsilon_i$, we get:
$\hat{\beta}_0 - \beta_0 = \frac{1}{n}\sum_{i=1}^{n} (\beta_0 + \beta_1 x_i + \epsilon_i) - \hat{\beta}_1 \bar{x} - \beta_0$
$= \beta_0 + \beta_1 \bar{x} + \frac{1}{n}\sum_{i=1}^{n} \epsilon_i - \hat{\beta}_1 \bar{x} - \beta_0$
$= \beta_1 \bar{x} + \frac{1}{n}\sum_{i=1}^{n} \epsilon_i - \hat{\beta}_1 \bar{x}$
$= (\beta_1 - \hat{\beta}_1)\bar{x} + \frac{1}{n}\sum_{i=1}^{n} \epsilon_i$
$= -(\hat{\beta}_1 - \beta_1)\bar{x} + \frac{1}{n}\sum_{i=1}^{n} \epsilon_i$

Step 5: Recall the formula for $\hat{\beta}_1$
$\hat{\beta}_1 - \beta_1 = \frac{S_{xy}}{S_{xx}} - \beta_1$

Substituting the model equation:
$S_{xy} = \sum_{i=1}^{n} (x_i - \bar{x})(y_i - \bar{y}) = \sum_{i=1}^{n} (x_i - \bar{x})(\beta_0 + \beta_1 x_i + \epsilon_i - \bar{y})$

Since $\bar{y} = \beta_0 + \beta_1 \bar{x} + \bar{\epsilon}$, we get:
$S_{xy} = \sum_{i=1}^{n} (x_i - \bar{x})(\beta_1(x_i - \bar{x}) + (\epsilon_i - \bar{\epsilon}))$
$= \beta_1 \sum_{i=1}^{n} (x_i - \bar{x})^2 + \sum_{i=1}^{n} (x_i - \bar{x})(\epsilon_i - \bar{\epsilon})$
$= \beta_1 S_{xx} + \sum_{i=1}^{n} (x_i - \bar{x})\epsilon_i$ (since $\sum_{i=1}^{n} (x_i - \bar{x}) = 0$)

Therefore:
$\hat{\beta}_1 - \beta_1 = \frac{\beta_1 S_{xx} + \sum_{i=1}^{n} (x_i - \bar{x})\epsilon_i}{S_{xx}} - \beta_1 = \frac{\sum_{i=1}^{n} (x_i - \bar{x})\epsilon_i}{S_{xx}}$

Step 6: Now we can compute the covariance
$Cov(\hat{\beta}_0, \hat{\beta}_1) = E[(\hat{\beta}_0 - \beta_0)(\hat{\beta}_1 - \beta_1)]$
$= E[(-(\hat{\beta}_1 - \beta_1)\bar{x} + \frac{1}{n}\sum_{i=1}^{n} \epsilon_i)(\frac{\sum_{j=1}^{n} (x_j - \bar{x})\epsilon_j}{S_{xx}})]$
$= E[(-\bar{x})(\frac{\sum_{i=1}^{n} (x_i - \bar{x})\epsilon_i}{S_{xx}})(\frac{\sum_{j=1}^{n} (x_j - \bar{x})\epsilon_j}{S_{xx}}) + (\frac{1}{n}\sum_{i=1}^{n} \epsilon_i)(\frac{\sum_{j=1}^{n} (x_j - \bar{x})\epsilon_j}{S_{xx}})]$

Step 7: Since $\epsilon_i$ are i.i.d. with variance $\sigma^2$, we have:
$E[\epsilon_i \epsilon_j] = \begin{cases} \sigma^2, & \text{if } i = j \\ 0, & \text{if } i \neq j \end{cases}$

Simplifying the first term:
$E[(\frac{\sum_{i=1}^{n} (x_i - \bar{x})\epsilon_i}{S_{xx}})^2] = \frac{E[\sum_{i=1}^{n}\sum_{j=1}^{n}(x_i - \bar{x})(x_j - \bar{x})\epsilon_i\epsilon_j]}{S_{xx}^2}$
$= \frac{\sum_{i=1}^{n}(x_i - \bar{x})^2\sigma^2}{S_{xx}^2} = \frac{\sigma^2 S_{xx}}{S_{xx}^2} = \frac{\sigma^2}{S_{xx}}$

The first term becomes: $-\bar{x} \cdot \frac{\sigma^2}{S_{xx}}$

For the second term:
$E[(\frac{1}{n}\sum_{i=1}^{n} \epsilon_i)(\frac{\sum_{j=1}^{n} (x_j - \bar{x})\epsilon_j}{S_{xx}})] = \frac{1}{n \cdot S_{xx}}E[\sum_{i=1}^{n}\sum_{j=1}^{n}\epsilon_i(x_j - \bar{x})\epsilon_j]$
$= \frac{1}{n \cdot S_{xx}}\sum_{i=1}^{n}(x_i - \bar{x})E[\epsilon_i^2] = \frac{1}{n \cdot S_{xx}}\sum_{i=1}^{n}(x_i - \bar{x})\sigma^2$
$= \frac{\sigma^2}{n \cdot S_{xx}}\sum_{i=1}^{n}(x_i - \bar{x}) = 0$ (since $\sum_{i=1}^{n}(x_i - \bar{x}) = 0$)

Step 8: Combining the terms, we get:
$Cov(\hat{\beta}_0, \hat{\beta}_1) = -\bar{x} \cdot \frac{\sigma^2}{S_{xx}} + 0 = -\frac{\bar{x} \cdot \sigma^2}{S_{xx}}$

Therefore: $Cov(\hat{\beta}_0, \hat{\beta}_1) = -\frac{\bar{x} \cdot \sigma^2}{S_{xx}}$

#### ✅ 1-b. $cov(\bar{y}, \hat{𝛽_1}) = 0$  

# Proving $ \text{cov}(\bar{y}, \hat{\beta}_1) = 0 $

We’re given the simple linear regression model:

$
y_i = \beta_0 + \beta_1 x_i + \epsilon_i, \quad i = 1, \dots, n
$

where $ \epsilon_i $ are i.i.d. $ N(\mu, \sigma^2) $. We also know:

$
\hat{\beta}_1 = \frac{S_{xy}}{S_{xx}} = \frac{\sum (x_i - \bar{x})(y_i - \bar{y})}{\sum (x_i - \bar{x})^2}
$

$
\hat{\beta}_0 = \bar{y} - \hat{\beta}_1 \bar{x}
$

and some key variance/covariance properties:

$
\text{Var}(\hat{\beta}_1) = \frac{\sigma^2}{S_{xx}}, \quad \text{Cov}(\hat{\beta}_0, \hat{\beta}_1) = -\frac{\bar{x} \sigma^2}{S_{xx}}
$

## Step 1: Express $ \bar{y} $ and $ \hat{\beta}_1 $

By definition:

$
\bar{y} = \frac{1}{n} \sum y_i
$

Expanding $ y_i $:

$
\bar{y} = \frac{1}{n} \sum (\beta_0 + \beta_1 x_i + \epsilon_i)
$

$
= \beta_0 + \beta_1 \bar{x} + \bar{\epsilon}
$

where $ \bar{\epsilon} = \frac{1}{n} \sum \epsilon_i $.

For $ \hat{\beta}_1 $, substituting $ y_i $ into $ S_{xy} $:

$
S_{xy} = \sum (x_i - \bar{x}) (\beta_0 + \beta_1 x_i + \epsilon_i - \bar{y})
$

$
= \sum (x_i - \bar{x}) (\beta_1 (x_i - \bar{x}) + \epsilon_i - \bar{\epsilon})
$

which simplifies to:

$
S_{xy} = \beta_1 S_{xx} + \sum (x_i - \bar{x})(\epsilon_i - \bar{\epsilon})
$

So:

$
\hat{\beta}_1 = \beta_1 + \frac{\sum (x_i - \bar{x})(\epsilon_i - \bar{\epsilon})}{S_{xx}}
$

## Step 2: Compute $ \text{cov}(\bar{y}, \hat{\beta}_1) $

By definition:

$
\text{cov}(\bar{y}, \hat{\beta}_1) = E[(\bar{y} - E[\bar{y}])(\hat{\beta}_1 - E[\hat{\beta}_1])]
$

We already found:

$
\bar{y} - E[\bar{y}] = \bar{\epsilon}, \quad \hat{\beta}_1 - E[\hat{\beta}_1] = \frac{\sum (x_i - \bar{x})(\epsilon_i - \bar{\epsilon})}{S_{xx}}
$

So:

$
\text{cov}(\bar{y}, \hat{\beta}_1) = E \left[ \bar{\epsilon} \cdot \frac{\sum (x_i - \bar{x})(\epsilon_i - \bar{\epsilon})}{S_{xx}} \right]
$

Since $ \bar{\epsilon} $ is independent of $ \sum (x_i - \bar{x})(\epsilon_i - \bar{\epsilon}) $ and both have an expected value of 0, we get:

$
\text{cov}(\bar{y}, \hat{\beta}_1) = 0
$

### Conclusion:

We’ve just shown that $ \bar{y} $ (the sample mean of $ y $) and $ \hat{\beta}_1 $ (the estimated slope) are uncorrelated:

$
\text{cov}(\bar{y}, \hat{\beta}_1) = 0
$

That’s it!


### 🤔 Question 2. (10%) 
Demonstrate that the regression sum of squares (SSR) can be computed using the following formula: $$ {SSR} = (\sum_{i=1,2,...n}{(\hat{y_i})^2}) - n * (\bar{y})^2 $$

## 📚 参考
 - [Datacamp SSR][1] (https://www.datacamp.com/tutorial/regression-sum-of-squares)

### 🚨 Page 28 of DA02 Regression Analysis.pdf
In linear regression, one LSE property is that $\sum_{i=1}^{n}{\hat{y}_i} = \sum_{i=1}^{n}{y_i}$

This implies that $\sum_{i=1}^{n}{\hat{y}_i} = n\bar{y}$

$SSR = \sum_{i=1}^{n}{(\hat{y}_i - \bar{y})^2}$

- $\hat{y}_i$ is the predicted value for observation $i$
- $\bar{y}$ is the mean of the observed values

$SSR = \sum_{i=1}^{n}{(\hat{y}_i - \bar{y})^2}$

$= \sum_{i=1}^{n}{(\hat{y}_i^2 - 2\hat{y}_i\bar{y} + \bar{y}^2)}$

$= \sum_{i=1}^{n}{\hat{y}_i^2} - \sum_{i=1}^{n}{2\hat{y}_i\bar{y}} + \sum_{i=1}^{n}{\bar{y}^2}$

$= \sum_{i=1}^{n}{\hat{y}_i^2} - 2\bar{y}\sum_{i=1}^{n}{\hat{y}_i} + \sum_{i=1}^{n}{\bar{y}^2}$

$= \sum_{i=1}^{n}{\hat{y}_i^2} - 2\bar{y}\sum_{i=1}^{n}{\hat{y}_i} + \bar{y}^2\sum_{i=1}^{n}{1}$

$= \sum_{i=1}^{n}{\hat{y}_i^2} - 2\bar{y}\sum_{i=1}^{n}{\hat{y}_i} + n\bar{y}^2$  
by 🚨, we get   
$= \sum_{i=1}^{n}{\hat{y}_i^2} - n\bar{y}^2$
## 👍 DONE.

### 🤔 Question 3. (10%)   
Given a multiple regression model: 𝐲 = 𝐗 * 𝛃 + 𝛜. Prove that the Least Squares Error (LSE) can be also expressed as
$\hat{\beta} = 𝛃 + R * 𝛜$  
where $𝐑 = ((𝐗^𝑇)*𝐗)^{−1}*(𝐗^𝑇)$.

Multiple regression model:
$\mathbf{y} = \mathbf{X}\boldsymbol{\beta} + \boldsymbol{\epsilon}$

The LSE (least squares estimator) $\hat{\boldsymbol{\beta}}$ minimizes the sum of squared residuals:
$\hat{\boldsymbol{\beta}} = \arg\min_{\boldsymbol{\beta}} (\mathbf{y} - \mathbf{X}\boldsymbol{\beta})^T(\mathbf{y} - \mathbf{X}\boldsymbol{\beta})$

Taking the derivative with respect to $\boldsymbol{\beta}$ and setting it to zero:
$-2\mathbf{X}^T(\mathbf{y} - \mathbf{X}\hat{\boldsymbol{\beta}}) = \mathbf{0}$

$-2\mathbf{X}^T(\mathbf{y} - \mathbf{X}\hat{\boldsymbol{\beta}}) = \mathbf{0}$  

$\mathbf{X}^T(\mathbf{y} - \mathbf{X}\hat{\boldsymbol{\beta}}) = \mathbf{0}$  

$\mathbf{X}^T \mathbf{y} - \mathbf{X}^T \mathbf{X}\hat{\boldsymbol{\beta}} = \mathbf{0}$  

$\mathbf{X}^T \mathbf{y} = \mathbf{X}^T \mathbf{X}\hat{\boldsymbol{\beta}}$  

Assuming $\mathbf{X}^T\mathbf{X}$ is invertible. $\Rightarrow \hat{\boldsymbol{\beta}} = (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{y}$

Subsituting multiple regression model $ \mathbf{y} = \mathbf{X}\boldsymbol{\beta} + \boldsymbol{\epsilon}$ into this expression
$\Rightarrow \hat{\boldsymbol{\beta}} = (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T(\mathbf{X}\boldsymbol{\beta} + \boldsymbol{\epsilon})$

$\hat{\boldsymbol{\beta}} = (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{X}\boldsymbol{\beta} + (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\boldsymbol{\epsilon}$

$ = \boldsymbol{\beta} + (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\boldsymbol{\epsilon}$

$ = \boldsymbol{\beta} + \mathbf{R}\boldsymbol{\epsilon}$
## 👍 DONE.

### 🤔 Question 4. (10%) 
The matrix, $𝐗*(((𝐗^𝑇)*𝐗)^{−1})*(𝐗^𝑇)$, derived in multiple regression is usually defined as 𝐇 and called the hat matrix.

#### ✅ 4-a. 𝐇 is idempotent, i.e., 𝐇*𝐇 = 𝐇 and (𝐈 − 𝐇)(𝐈 − 𝐇) = (𝐈 − 𝐇) 

$\mathbf{H} = \mathbf{X}((\mathbf{X}^T\mathbf{X})^{-1})\mathbf{X}^T$

#### ✅ 4-a-a: Proving $\mathbf{H}$ is idempotent, i.e., $\mathbf{H}\mathbf{H} = \mathbf{H}$

$\mathbf{H}\mathbf{H} = \mathbf{X}((\mathbf{X}^T\mathbf{X})^{-1})\mathbf{X}^T \cdot \mathbf{X}((\mathbf{X}^T\mathbf{X})^{-1})\mathbf{X}^T$

$\mathbf{H}\mathbf{H} = \mathbf{X}((\mathbf{X}^T\mathbf{X})^{-1}) \cdot \mathbf{X}^T\mathbf{X} \cdot ((\mathbf{X}^T\mathbf{X})^{-1})\mathbf{X}^T$

$\mathbf{H}\mathbf{H} = \mathbf{X}((\mathbf{X}^T\mathbf{X})^{-1}) \cdot \mathbf{X}^T$

$\mathbf{H}\mathbf{H} = \mathbf{H}$

## 👍 DONE.

#### ✅ 4-a-b: Proving $(\mathbf{I} - \mathbf{H})(\mathbf{I} - \mathbf{H}) = (\mathbf{I} - \mathbf{H})$

$(\mathbf{I} - \mathbf{H})(\mathbf{I} - \mathbf{H}) = \mathbf{I} - \mathbf{H} - \mathbf{H} + \mathbf{H}\mathbf{H}$

$(\mathbf{I} - \mathbf{H})(\mathbf{I} - \mathbf{H}) = \mathbf{I} - \mathbf{H} - \mathbf{H} + \mathbf{H}$

$(\mathbf{I} - \mathbf{H})(\mathbf{I} - \mathbf{H}) = \mathbf{I} - \mathbf{H}$

## 👍 DONE.

#### ✅ 4-b. $V(\hat{𝐲})= (𝜎^2) * 𝐇$  

In multiple regression, we have:
- $\mathbf{y}$ is the vector of observed responses
- $\hat{\mathbf{y}}$ is the vector of predicted values
- $\mathbf{X}$ is the design matrix
- $\boldsymbol{\beta}$ is the vector of regression coefficients
- $\boldsymbol{\varepsilon}$ is the vector of errors
- $\sigma^2$ is the variance of the error terms

linear regression model:
$\mathbf{y} = \mathbf{X}\boldsymbol{\beta} + \boldsymbol{\varepsilon}$

The ordinary least squares (OLS) estimator for $\boldsymbol{\beta}$ is:
$\hat{\boldsymbol{\beta}} = (\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{y}$

The predicted values $\hat{\mathbf{y}}$ are given by:
$\hat{\mathbf{y}} = \mathbf{X}\hat{\boldsymbol{\beta}}$

$\Rightarrow \hat{\mathbf{y}} = \mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{y} = \mathbf{H}\mathbf{y}$


$\Rightarrow V(\hat{\mathbf{y}}) = V(\mathbf{H}\mathbf{y})$  
$\Rightarrow V(\hat{\mathbf{y}}) = \mathbf{H} \cdot V(\mathbf{y}) \cdot \mathbf{H}^T$

$\Rightarrow V(\hat{\mathbf{y}}) = \mathbf{H} \cdot (V(\mathbf{X}\boldsymbol{\beta} + \boldsymbol{\varepsilon})) \cdot \mathbf{H}^T$

$\Rightarrow V(\hat{\mathbf{y}}) = \mathbf{H} \cdot (V(\boldsymbol{\varepsilon})) \cdot \mathbf{H}^T$

$\Rightarrow V(\hat{\mathbf{y}}) = \mathbf{H} \cdot (\sigma^2\mathbf{I}) \cdot \mathbf{H}^T$

$\Rightarrow V(\hat{\mathbf{y}}) = \mathbf{H} \cdot \sigma^2\mathbf{I} \cdot \mathbf{H}^T$

$\Rightarrow V(\hat{\mathbf{y}}) = \sigma^2 \cdot \mathbf{H} \cdot \mathbf{H}^T$

Since the hat matrix $\mathbf{H}$ is symmetric, $\mathbf{H}^T = \mathbf{H}$.

Therefore:
$V(\hat{\mathbf{y}}) = \sigma^2 \cdot \mathbf{H} \cdot \mathbf{H}$

And since we proved in part (a) that $\mathbf{H}$ is idempotent (i.e., $\mathbf{H}\mathbf{H} = \mathbf{H}$):

$V(\hat{\mathbf{y}}) = \sigma^2 \cdot \mathbf{H}$

## 👍 DONE.

### 🤔 Question 5. (20%) 

Given a simple linear regression model,  

#### ✅ 5-a. show that the elements of the hat matrix can be expressed as:
$h_{ii} = \frac{1}{n} + \frac{ ( x_{i} - \bar{x} )^2}{ S_{xx} }$  
$h_{ij} = \frac{1}{n} + \frac{ ( x_{i} - \bar{x} )( x_{j} - \bar{x} )}{ S_{xx}}$

# Hat Matrix Elements in Simple Linear Regression

## Step 1: Define the linear regression model
Our simple linear regression model is:
$$y_i = \beta_0 + \beta_1 x_i + \varepsilon_i$$

In matrix form, this is:
$$\mathbf{y} = \mathbf{X} \mathbf{\beta} + \mathbf{\varepsilon}$$

where $\mathbf{X}$ is an $n \times 2$ matrix with a column of 1's and a column of $x$ values.

## Step 2: Define the hat matrix
The hat matrix $\mathbf{H}$ maps the observed values $\mathbf{y}$ to the fitted values $\mathbf{\hat{y}}$:
$$\mathbf{\hat{y}} = \mathbf{H} \mathbf{y}$$

The hat matrix is defined as:
$$\mathbf{H} = \mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T$$

## Step 3: Expand the design matrix $\mathbf{X}$
$$\mathbf{X} = \begin{bmatrix} 
1 & x_1 \\
1 & x_2 \\
\vdots & \vdots \\
1 & x_n
\end{bmatrix}$$

## Step 4: Calculate $\mathbf{X}^T\mathbf{X}$
$$\mathbf{X}^T\mathbf{X} = \begin{bmatrix} 
1 & 1 & \cdots & 1 \\
x_1 & x_2 & \cdots & x_n
\end{bmatrix}
\begin{bmatrix} 
1 & x_1 \\
1 & x_2 \\
\vdots & \vdots \\
1 & x_n
\end{bmatrix} = 
\begin{bmatrix} 
n & \sum_{i=1}^n x_i \\
\sum_{i=1}^n x_i & \sum_{i=1}^n x_i^2
\end{bmatrix}$$

## Step 5: Calculate $(\mathbf{X}^T\mathbf{X})^{-1}$
Let's denote $\sum_{i=1}^n x_i = n\bar{x}$ and $S_{xx} = \sum_{i=1}^n (x_i - \bar{x})^2 = \sum_{i=1}^n x_i^2 - n\bar{x}^2$.

The inverse of a $2 \times 2$ matrix $\begin{bmatrix} a & b \\ c & d \end{bmatrix}$ is $\frac{1}{ad-bc}\begin{bmatrix} d & -b \\ -c & a \end{bmatrix}$.

Applying this:
$$(\mathbf{X}^T\mathbf{X})^{-1} = \frac{1}{n\sum_{i=1}^n x_i^2 - (\sum_{i=1}^n x_i)^2}
\begin{bmatrix} 
\sum_{i=1}^n x_i^2 & -\sum_{i=1}^n x_i \\
-\sum_{i=1}^n x_i & n
\end{bmatrix}$$

Substituting our notations:
$$(\mathbf{X}^T\mathbf{X})^{-1} = \frac{1}{nS_{xx} + n^2\bar{x}^2 - n^2\bar{x}^2}
\begin{bmatrix} 
S_{xx} + n\bar{x}^2 & -n\bar{x} \\
-n\bar{x} & n
\end{bmatrix} = 
\frac{1}{nS_{xx}}
\begin{bmatrix} 
S_{xx} + n\bar{x}^2 & -n\bar{x} \\
-n\bar{x} & n
\end{bmatrix}$$

## Step 6: Calculate $\mathbf{H} = \mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T$
Consider the $i$-th row of $\mathbf{X}$, which is $[1 \, x_i]$, and the $j$-th column of $\mathbf{X}^T$, which is $[1 \, x_j]^T$.

The $(i,j)$ element of $\mathbf{H}$ is:
$$h_{ij} = [1 \, x_i] \cdot (\mathbf{X}^T\mathbf{X})^{-1} \cdot [1 \, x_j]^T$$

Expanding this with the inverse we calculated:
$$h_{ij} = [1 \, x_i] \cdot \frac{1}{nS_{xx}}
\begin{bmatrix} 
S_{xx} + n\bar{x}^2 & -n\bar{x} \\
-n\bar{x} & n
\end{bmatrix} \cdot [1 \, x_j]^T$$

$$h_{ij} = \frac{1}{nS_{xx}} \left[ (S_{xx} + n\bar{x}^2) - n\bar{x}x_i - n\bar{x}x_j + nx_ix_j \right]$$

## Step 7: Simplify the expression for $h_{ij}$
$$h_{ij} = \frac{1}{nS_{xx}} \left[ S_{xx} + n\bar{x}^2 - n\bar{x}x_i - n\bar{x}x_j + nx_ix_j \right]$$

$$h_{ij} = \frac{S_{xx}}{nS_{xx}} + \frac{n\bar{x}^2}{nS_{xx}} - \frac{n\bar{x}x_i}{nS_{xx}} - \frac{n\bar{x}x_j}{nS_{xx}} + \frac{nx_ix_j}{nS_{xx}}$$

$$h_{ij} = \frac{1}{n} + \frac{\bar{x}^2}{S_{xx}} - \frac{\bar{x}x_i}{S_{xx}} - \frac{\bar{x}x_j}{S_{xx}} + \frac{x_ix_j}{S_{xx}}$$

Rearranging terms:
$$h_{ij} = \frac{1}{n} + \frac{1}{S_{xx}}(\bar{x}^2 - \bar{x}x_i - \bar{x}x_j + x_ix_j)$$

$$h_{ij} = \frac{1}{n} + \frac{1}{S_{xx}}(x_ix_j - \bar{x}x_i - \bar{x}x_j + \bar{x}^2)$$

This can be rewritten as:
$$h_{ij} = \frac{1}{n} + \frac{(x_i - \bar{x})(x_j - \bar{x})}{S_{xx}}$$

## Step 8: Special case for $h_{ii}$ (diagonal elements)
For the diagonal elements where $i = j$:
$$h_{ii} = \frac{1}{n} + \frac{(x_i - \bar{x})(x_i - \bar{x})}{S_{xx}} = \frac{1}{n} + \frac{(x_i - \bar{x})^2}{S_{xx}}$$

## Summary
Therefore, the elements of the hat matrix can be expressed as:
- Diagonal elements: $h_{ii} = \frac{1}{n} + \frac{(x_i - \bar{x})^2}{S_{xx}}$
- Off-diagonal elements: $h_{ij} = \frac{1}{n} + \frac{(x_i - \bar{x})(x_j - \bar{x})}{S_{xx}}$

## 👍 DONE.

### ✅ 5-b-1. Behavior of Diagonal Elements $h_{ii}$

The diagonal element $h_{ii}$ is given by:
$$h_{ii} = \frac{1}{n} + \frac{(x_i - \bar{x})^2}{S_{xx}}$$

Where:
- $\frac{1}{n}$ is a constant term
- $\frac{(x_i - \bar{x})^2}{S_{xx}}$ depends on how far $x_i$ is from the mean $\bar{x}$

As $x_i$ deviates from $\bar{x}$:
1. When $x_i = \bar{x}$:
   - $(x_i - \bar{x})^2 = 0$
   - $h_{ii} = \frac{1}{n}$ (minimum value)

2. When $x_i$ moves away from $\bar{x}$ in either direction:
   - $(x_i - \bar{x})^2$ increases quadratically
   - $h_{ii}$ increases quadratically

3. When $x_i$ is far from $\bar{x}$:
   - $h_{ii}$ becomes larger, approaching 1 for extreme values
   - $h_{ii} \to 1$ as $|x_i - \bar{x}| \to \text{extreme}$

This means observations with $x$ values far from the mean have higher "leverage" in the regression. These high-leverage points have greater influence on the fitted regression line, as their $h_{ii}$ values indicate how much the fitted value $\hat{y}_i$ depends on the observed value $y_i$.


### ✅ 5-b-2. Behavior of Off-Diagonal Elements $h_{ij}$

The off-diagonal element $h_{ij}$ is given by:
$$h_{ij} = \frac{1}{n} + \frac{(x_i - \bar{x})(x_j - \bar{x})}{S_{xx}}$$

The behavior of $h_{ij}$ depends on both $x_i$ and $x_j$ relative to $\bar{x}$:

1. When either $x_i = \bar{x}$ or $x_j = \bar{x}$:
   - $(x_i - \bar{x})(x_j - \bar{x}) = 0$
   - $h_{ij} = \frac{1}{n}$

2. When $x_i$ and $x_j$ are on the same side of $\bar{x}$ (both above or both below):
   - $(x_i - \bar{x})(x_j - \bar{x}) > 0$ (positive)
   - $h_{ij} > \frac{1}{n}$ (larger than the baseline)

3. When $x_i$ and $x_j$ are on opposite sides of $\bar{x}$:
   - $(x_i - \bar{x})(x_j - \bar{x}) < 0$ (negative)
   - $h_{ij} < \frac{1}{n}$ (smaller than the baseline)
   - In extreme cases, $h_{ij}$ can become negative

This indicates how observations with $x$ values influence each other's fitted values. Observations on the same side of the mean tend to positively influence each other's predictions, while observations on opposite sides of the mean can negatively influence each other's predictions.

## Implications

1. **Leverage points**: Observations with $x$ values far from $\bar{x}$ have high $h_{ii}$ values and thus high leverage, making them potentially influential in determining the regression line.

2. **Influence distribution**: As the leverage increases for observations far from the mean, the influence of observations near the mean decreases.

3. **Outlier sensitivity**: The quadratic relationship means that extreme $x$ values have disproportionately high influence, making the regression sensitive to outliers in the predictor space.

4. **Sum constraint**: The sum of all $h_{ii}$ equals 2 for simple linear regression, so if some observations gain leverage, others must lose it.

5. **Model interpretation**: The hat matrix helps identify which observations are most influential in determining the fitted regression line, which is valuable for diagnostics and outlier detection.

In practice, observations with $h_{ii} > \frac{2p}{n}$ (where $p$ is the number of parameters, here $p=2$) are often considered high-leverage points that warrant special attention in regression diagnostics.
## 👍 DONE.

### 🤔 Question 6. (20%) 

Assuming that the true underlying model is
$𝐲 = {\beta}_0 + {\beta}_1 * 𝐱_1 + {\beta}_2 * 𝐱_2 + ⋯ {\beta}_p * 𝐱_𝑝 + 𝛜 = 𝐗 * {\beta} + 𝛜,$

If we intentionally ignores the intercept term and fit the data with the following model:  
$𝐲 = \hat{\beta}_1 * 𝐱_1 + \hat{\beta}_2 * 𝐱_2 + ⋯ \hat{\beta}_p * 𝐱_𝑝 = 𝐗_{prime} * \hat{\beta}_{prim}$

### ✅ 6-a. (10%) Show that 

$E[\hat{{\beta}}_1] \neq \beta_1$
$E[\hat{{\beta}}_2] \neq \beta_2$
...
$E[\hat{{\beta}}_p] \neq \beta_p$

## True Model vs. Fitted Model

- **True Model**: $\mathbf{y} = \beta_0 + \beta_1\mathbf{x}_1 + \beta_2\mathbf{x}_2 + \cdots + \beta_p\mathbf{x}_p + \boldsymbol{\epsilon} = \mathbf{X}\boldsymbol{\beta} + \boldsymbol{\epsilon}$
  
  Where $\mathbf{X} = [\mathbf{1}, \mathbf{x}_1, \mathbf{x}_2, \ldots, \mathbf{x}_p]$ and $\boldsymbol{\beta} = [\beta_0, \beta_1, \beta_2, \ldots, \beta_p]^T$

- **Fitted Model**: $\mathbf{y} = \hat{\beta}_1\mathbf{x}_1 + \hat{\beta}_2\mathbf{x}_2 + \cdots + \hat{\beta}_p\mathbf{x}_p = \mathbf{X}_{prime}\hat{\boldsymbol{\beta}}_{prime}$
  
  Where $\mathbf{X}_{prime} = [\mathbf{x}_1, \mathbf{x}_2, \ldots, \mathbf{x}_p]$ and $\hat{\boldsymbol{\beta}}_{prime} = [\hat{\beta}_1, \hat{\beta}_2, \ldots, \hat{\beta}_p]^T$

## Parameter Estimation

For the fitted model, the least squares estimate is:
$$\hat{\boldsymbol{\beta}}_{prime} = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{y}$$

Substituting the true model:
$$\hat{\boldsymbol{\beta}}_{prime} = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T(\mathbf{X}\boldsymbol{\beta} + \boldsymbol{\epsilon})$$

$$\hat{\boldsymbol{\beta}}_{prime} = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{X}\boldsymbol{\beta} + (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\boldsymbol{\epsilon}$$

## Expected Value

Taking the expected value:
$$E[\hat{\boldsymbol{\beta}}_{prime}] = E[(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{X}\boldsymbol{\beta} + (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\boldsymbol{\epsilon}]$$

Since $E[\boldsymbol{\epsilon}] = \mathbf{0}$:
$$E[\hat{\boldsymbol{\beta}}_{prime}] = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{X}\boldsymbol{\beta}$$

Now, let's expand this. We have:
$$\mathbf{X} = [\mathbf{1}, \mathbf{x}_1, \mathbf{x}_2, \ldots, \mathbf{x}_p]$$
$$\boldsymbol{\beta} = [\beta_0, \beta_1, \beta_2, \ldots, \beta_p]^T$$

So:
$$\mathbf{X}\boldsymbol{\beta} = \beta_0\mathbf{1} + \beta_1\mathbf{x}_1 + \beta_2\mathbf{x}_2 + \cdots + \beta_p\mathbf{x}_p$$

Substituting back:
$$E[\hat{\boldsymbol{\beta}}_{prime}] = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T(\beta_0\mathbf{1} + \beta_1\mathbf{x}_1 + \beta_2\mathbf{x}_2 + \cdots + \beta_p\mathbf{x}_p)$$

$$E[\hat{\boldsymbol{\beta}}_{prime}] = (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\beta_0\mathbf{1} + (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T(\beta_1\mathbf{x}_1 + \beta_2\mathbf{x}_2 + \cdots + \beta_p\mathbf{x}_p)$$

$$E[\hat{\boldsymbol{\beta}}_{prime}] = \beta_0(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{1} + (\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{X}_{prime}[\beta_1, \beta_2, \ldots, \beta_p]^T$$

Since $(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{X}_{prime} = \mathbf{I}$ (identity matrix):

$$E[\hat{\boldsymbol{\beta}}_{prime}] = \beta_0(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{1} + [\beta_1, \beta_2, \ldots, \beta_p]^T$$

## The Key Point

The critical term here is $\beta_0(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{1}$. Unless all predictors are centered (i.e., all $\mathbf{x}_i$ have mean 0), this term will not be zero.

Let's denote this bias term as $\mathbf{b} = \beta_0(\mathbf{X}_{prime}^T\mathbf{X}_{prime})^{-1}\mathbf{X}_{prime}^T\mathbf{1}$.

Then:
$$E[\hat{\boldsymbol{\beta}}_{prime}] = \mathbf{b} + [\beta_1, \beta_2, \ldots, \beta_p]^T$$

For each individual coefficient:
$$E[\hat{\beta}_1] = b_1 + \beta_1 \neq \beta_1 \text{ (unless } b_1 = 0\text{)}$$
$$E[\hat{\beta}_2] = b_2 + \beta_2 \neq \beta_2 \text{ (unless } b_2 = 0\text{)}$$
$$\vdots$$
$$E[\hat{\beta}_p] = b_p + \beta_p \neq \beta_p \text{ (unless } b_p = 0\text{)}$$

The bias terms $(b_1, b_2, \ldots, b_p)$ will only be zero if:
1. $\beta_0 = 0$ (the true intercept is zero), or
2. $\mathbf{X}_{prime}^T\mathbf{1} = \mathbf{0}$ (all predictors are centered, i.e., have mean zero)

Since we're not assuming either of these conditions, we have shown that ignoring the intercept term when it exists in the true model leads to biased estimates of all regression coefficients:

$$E[\hat{\beta}_j] \neq \beta_j \text{ for } j = 1, 2, \ldots, p$$
## 👍 DONE.

### ✅ 6-b. (5%) What can you conclude from (a)?
All estimates are **BIASED** when we omit the intercept term from a model where it truly exists.
## 👍 DONE.

### 🤔 Question 7. (15%) 

In a multiple regression model: $𝐲 = 𝐗 * \beta + 𝛜$, it is critical to know if $((𝐗^𝑇)*𝐗)^{−1}$ exists.  
The diagonal elements of $((𝐗^𝑇)*𝐗)^{−1}$ in the correlation form, where 𝐗 is standardized, are known as Variance Inflation Factors
(VIFs). They are crucial to diagnose multicollinearity. VIF for the 𝑗-𝑡ℎ regression coefficient is expressed as

${VIF}_{j} = \frac{1}{1-(R_j)^2}$

$R_j$ is the coefficient of multiple determination obtained from regressing 𝐱_𝑗 on the other regressor variables (𝐱_1 to 𝐱_𝑝, except 𝐱_𝑗). Calculate all the VIFs in the “autompg” dataset and discuss your observation.

In [3]:
# 必要なライブラリを再ロード
import pandas as pd
import numpy as np
from sklearn.preprocessing import StandardScaler
from statsmodels.stats.outliers_influence import variance_inflation_factor

# ファイルパス
filePath = "./auto-mpg.data.txt"

# データセットの説明に基づく列名
columnNames = [
    "mpg", "cylinders", "displacement", "horsepower", "weight", 
    "acceleration", "model_year", "origin", "car_name"
]

# データセットをロード
df = pd.read_csv(filePath, delim_whitespace=True, names=columnNames, na_values='?')

# 'horsepower' の欠損値を持つ行を削除
df.dropna(inplace=True)

# 'horsepower' を float に変換（欠損値のために文字列として読み込まれた）
df['horsepower'] = df['horsepower'].astype(float)

# 'car_name' は数値ではないため削除
df.drop(columns=['car_name'], inplace=True)

# 独立変数（'mpg' は従属変数なので除外）を定義
X = df.drop(columns=['mpg'])

# 独立変数を標準化（相関形式）
scaler = StandardScaler()
xScaled = scaler.fit_transform(X)

# 各独立変数の VIF を計算
vifs = pd.DataFrame()
vifs["Variable"] = X.columns
vifs["VIF"] = [variance_inflation_factor(xScaled, i) for i in range(xScaled.shape[1])]

# 計算された VIF を表示
vifs

Variable        VIF
0     cylinders  10.737535
1  displacement  21.836792
2    horsepower   9.943693
3        weight  10.831260
4  acceleration   2.625806
5    model_year   1.244952
6        origin   1.772386

$
\begin{array}{|c|c|}
\hline
\text{Variable} & \text{VIF} \\
\hline
\text{cylinders} & 10.74 \\
\text{displacement} & 21.84 \\
\text{horsepower} & 9.94 \\
\text{weight} & 10.83 \\
\text{acceleration} & 2.63 \\
\text{model\_year} & 1.24 \\
\text{origin} & 1.77 \\
\hline
\end{array}
$

## **VIF分析（auto-mpgデータセット）からの結論：**

* **高い多重共線性：**
    * "cylinders"、"displacement"、"horsepower"、"weight"は高いVIF値（5または10より大きい）を示す。
    * これらの変数間に強い線形関係が存在する。
* **中程度の多重共線性：**
    * "acceleration"は中程度の多重共線性を示す。
* **低い多重共線性：**
    * "model_year"と"origin"は低い多重共線性を示す。
* **影響：**
    * 高い相関を持つ変数の係数推定値は信頼性が低い。
    * 個々の変数の影響を評価することが困難。

### **Observations:**
1. **High Multicollinearity:**  
   - `displacement` (VIF ≈ 21.84), `weight` (VIF ≈ 10.83), and `cylinders` (VIF ≈ 10.74) have high VIF values, indicating strong multicollinearity.  
   - This suggests that these variables are highly correlated with each other. Likely, `displacement`, `cylinders`, and `weight` are strongly related since larger engines tend to have more cylinders and weigh more.

2. **Moderate Multicollinearity:**  
   - `horsepower` (VIF ≈ 9.94) also exhibits some collinearity, which might be due to its correlation with engine displacement and weight.

3. **Low Multicollinearity:**  
   - `acceleration` (VIF ≈ 2.63) shows some correlation but is not extreme.
   - `model_year` (VIF ≈ 1.24) and `origin` (VIF ≈ 1.77) have low VIF values, meaning they are relatively independent.

### **Conclusion:**
- High VIF values suggest that some predictors (`displacement`, `cylinders`, `weight`, and `horsepower`) may need to be removed or transformed to reduce multicollinearity.
- One approach to handling this would be **Principal Component Analysis (PCA)** or **removing redundant variables** to improve model interpretability.